helix-core/src/test/java/org/apache/helix/controller/rebalancer/waged/constraints/TestInstancePartitionsCountConstraint.java

```diff
@@ -28,36 +28,69 @@
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
+/**
+ * Unit test for {@link InstancePartitionsCountConstraint}
+ * The scores in the test method are pre-calculated and verified, any changes to the score method in
+ * the future needs to be aware of the test result
+ */
 public class TestInstancePartitionsCountConstraint {
   private final AssignableReplica _testReplica = Mockito.mock(AssignableReplica.class);
   private final AssignableNode _testNode = Mockito.mock(AssignableNode.class);
   private final ClusterContext _clusterContext = Mockito.mock(ClusterContext.class);
-
   private final SoftConstraint _constraint = new InstancePartitionsCountConstraint();
 
   @Test
-  public void testWhenInstanceIsIdle() {
+  public void testWhenZeroUsage() {
     when(_testNode.getAssignedReplicaCount()).thenReturn(0);
-    float score =
+    when(_clusterContext.getEstimatedMaxPartitionCount()).thenReturn(10);
+    float score = _constraint.getAssignmentScore(_testNode, _testReplica, _clusterContext);
+    float normalizedScore =
+        _constraint.getAssignmentNormalizedScore(_testNode, _testReplica, _clusterContext);
+    Assert.assertEquals(score, 0f);
+    Assert.assertEquals(normalizedScore, 1f);
+  }
+
+  @Test
+  public void testWhenExceedUsage() {
+    when(_testNode.getAssignedReplicaCount()).thenReturn(20);
+    when(_clusterContext.getEstimatedMaxPartitionCount()).thenReturn(10);
+    float score = _constraint.getAssignmentScore(_testNode, _testReplica, _clusterContext);
+    float normalizedScore =
         _constraint.getAssignmentNormalizedScore(_testNode, _testReplica, _clusterContext);
-    Assert.assertEquals(score, 1.0f);
+    Assert.assertEquals(score, 2f);
+    Assert.assertEquals(normalizedScore, 4.9999997E-4f);
```

Comment:
Probably a good idea to add some description in front of the method. Also if you could explain what these values all mean so that the reader could understand. For example, a 3rd person reviewing the code won't have a clue what "4.9999997E-4f" means here and why it has to be that value :)

Reply:
The value is just a record of the normalized score, we don't need to understand the absolute value it's meaningless. The unit test serves two purposes in the PR:
1. A record of pre-computed value given the input. So that if anyone touches the scoring method and gets things changed, he would be aware of it cause the output value is different.
2. The test method tests score when 0% usage, 1% usage, 50% usage, 100%, and 200% usage of some "aspect". From the tests, we can see the final normalized scores are decreasing and in range (0, 1] that's good enough.